### PR TITLE
refactor: Migrate to IntelliJ Platform Plugin

### PR DESCRIPTION
**Pull request by RehanChalana**
- use new IntellijPlatform syntax in build.gradle.kts
- bump to lsp4ij:0.6.0
- bump to gradleVersion 8.10.0

#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

build.gradle.kts
gradle.properties
gradle/libs.versions.toml

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [x] Manual test (add detailed scripts or steps below)
- [ ] Other

I have tried migrating from gradle Intellij plugin (1.x) to newer IntelliJ platform Plugin. But as i am not good with gradle tasks, i hope someone can please look into that and help me point in the right direction.

We should also add gradle properties like sinceBuild and updateBuild i am not really sure about them.

I was also trying to add the new syntax for "runIdeForUiTest" task, but again i am not sure about that.

So yes this pr really needs a review from someone !.

Also can you please give it a "hacktoberfest-accepted" tag, it will help me complete the hacktober-fest challenge

### CODE REVIEW

**Comment by RehanChalana** [comment]
@He1pa @Peefy can you please look into it.

**Comment by He1pa** [comment]
Do you have any documentation about sinceBuild and updateBuild or where did you refer to this migration? Previously we had 
```
pluginSinceBuild = 221
 # pluginUntilBuild = 233.*
``` 
 in gradle.properties. It means supported build number ranges and IntelliJ Platform versions 
We set a minimum version but no maximum version to be compatible with subsequent new versions. 
I am not sure if this means the same property

File: build.gradle.kts [hunk 1]
```diff
@@ -98,17 +83,42 @@ tasks {
                 )
             }
         }
+
+        ideaVersion {
+            sinceBuild = providers.gradleProperty("pluginSinceBuild")
+            untilBuild = providers.gradleProperty("pluginUntilBuild")
```
Comment (He1pa) [inline build.gradle.kts:89]: Do we need untilBuild?

File: README.md [hunk 2]
```diff
@@ -31,8 +34,15 @@ This extension provides many coding assistance based on Language Server Protocol
 + **Code Action:** Quick fix for some errors
 + **InlayHint:** Hint for variable type, function and schema args
 
+![image](https://github.com/user-attachments/assets/b0e4af8b-f4dc-4e93-ac05-b94020ba43dc)
```
Comment (He1pa) [inline README.md:37]: save as static image

**Review by He1pa — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (RehanChalana) [reply build.gradle.kts]: I have removed it

File: README.md (hunk 2 above)
Comment (RehanChalana) [reply README.md]: done

**Review by He1pa — APPROVED** [review]
LGTM

**Pull request merged**